lib/spack/spack/package.py

```diff
@@ -974,8 +974,8 @@ def do_fetch(self, mirror_only=False):
                 raise FetchError("Will not fetch %s" %
                                  self.spec.format('$_$@'), ck_msg)
 
+        self.stage.create()
```

Comment:
Is `do_fetch` always granted to happen before any of the other operations listed above (`stage`, `patch`, `install`)? Just trying to figure out if we have corner cases where moving the create will cause something to fail...

Reply:
Yes, it’s always called indirectly through do_{patch, stage, install}. That was originally most of the methods in Package, so lazy stage creation kind of made sense. Now it just makes the code less explicit and harder to understand where directories might’ve popped up.